pkg/network/controllers/vm_test.go

```diff
@@ -509,6 +509,70 @@ var _ = Describe("VM Network Controller", func() {
 		Expect(iface).NotTo(BeNil())
 		Expect(iface.State).NotTo(Equal(v1.InterfaceStateAbsent))
 	})
+
+	It("Sync successfully performs hotplug and hotunplug at the same time", func() {
+		const (
+			netToAttachName = "attach-me"
+
+			netToDetachName    = "detach-me"
+			netToDetachNADName = "detach-me-nad"
+		)
+
+		clientset := fake.NewSimpleClientset()
+		c := controllers.NewVMController(clientset)
+
+		multusAndDomainInfoSource := vmispec.NewInfoSource(vmispec.InfoSourceMultusStatus, vmispec.InfoSourceDomain)
+
+		ifaceToDetach := libvmi.InterfaceDeviceWithBridgeBinding(netToDetachName)
+		netToDetach := libvmi.MultusNetwork(netToDetachName, netToDetachNADName)
+
+		vmi := libvmi.New(
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+			libvmi.WithInterface(ifaceToDetach),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			libvmi.WithNetwork(netToDetach),
+			libvmistatus.WithStatus(
+				libvmistatus.New(
+					libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+						Name:             defaultNetName,
+						PodInterfaceName: namescheme.PrimaryPodInterfaceName,
+						InfoSource:       vmispec.InfoSourceDomain,
+					}),
+					libvmistatus.WithInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{
+						Name:             netToDetachName,
+						PodInterfaceName: namescheme.GenerateHashedInterfaceName(netToDetachName),
+						InfoSource:       multusAndDomainInfoSource,
+					}),
+				),
+			),
+		)
+
+		vm := libvmi.NewVirtualMachine(vmi.DeepCopy())
+
+		// Mark the secondary interface for hotunplug
+		vm.Spec.Template.Spec.Domain.Devices.Interfaces[1].State = v1.InterfaceStateAbsent
+
+		vm = plugNetworkInterface(vm, libvmi.InterfaceDeviceWithBridgeBinding(netToAttachName))
+
+		// Simulate the existence of the VMI on the server (to allow the Sync to patch it).
+		_, err := clientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Create(context.Background(), vmi, k8smetav1.CreateOptions{})
```

Comment:
I think the test missing the VirtualMachine creation as well, having the test simulate VM existence should prevent unwanted side effects.

Reply:
The VM objects are not "created" on the fake client all over the file.
This is because the network VM controller does not directly fetches / mutates the VirtualMachine object (currently it just patches the VMI).